designs/NuGet-ClientCertificates.md

```diff
@@ -0,0 +1,324 @@
+
+```xml
+<configuration>
+    ...
+    <packageSources>
+        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+        <add key="Contoso" value="https://contoso.com/packages/" />
+        <add key="Foo" value="https://foo.com/bar/packages/" />
+    </packageSources>
+    ...
+    <clientCertificates>	
+        <storeCert packageSource="Contoso"
+                   storeLocation="currentUser"
+                   storeName="my"
+                   findBy="thumbprint" 
+                   findValue="4894671ae5aa84840cc1079e89e82d426bc24ec6" />
+        <fileCert packageSource="Foo"
+                  path=".\certificate.pfx" 
+                  password="..." />
+    </clientCertificates>
+...
+</configuration>
+```
+# NuGet.exe
+
+## client-certificates command
+
+Gets, updates, sets or lists client certificates to the NuGet configuration.
+
+Usage
+```
+nuget client-certificates <list|add|remove|update> [options]
+```
+
+if none of `list|add|remove|update` is specified, the command will default to `list`.
+
+### nuget client-certificates list [options]
+
+Lists all the client certificates in the configuration. This option will include all configured client certificates that match to specified options.
+
+- `-PackageSource` - Filter available client certificates for specific source.
+
+- `-SourceType` one of `file|storage` - Filter available client certificates by it's source type.
+
+- `-Name` `string` - Filter client certificates **from all sources** by string presence in package source name.
+
+- `-Path` `string` - Filter client certificates **from file source** by string presence in it's Path.
+
+- `-StoreLocation` [possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storelocation?view=netframework-4.8) - Filter client certificates **from storage source** by it's StoreLocation.
+
+- `-StoreName` [possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storename?view=netframework-4.8) - Filter client certificates **from storage source** by it's StoreName.
+
+- `-FindBy` [possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509findtype?view=netframework-4.8) - Filter client certificates **from storage source** by it's FindBy.
+
+- `-FindValue` `string` - Filter client certificates from storage source by string presence in it's FindValue.
+
+Below is an example output from this command:
+
+```
+Registered client certificates:
+
+ 1.   Contoso [fileCert]
+      Path: d:\Temp\nuget\foo.pfx
+      Password: ****
+      Certificate: 4894671AE5AA84840C31079E89E82D426BC24EC6
+
+ 2.   Foo [storeCert]
+      Store location: CurrentUser
+      Store name: My
+      Find by: Thumbprint
+      Find value: ba4d3cc1f011388626a23af5627bb4721d44db6e
+      Certificate: BA4D3CC1F011388626A23AF5627BB4721D44DB6E
+      
+ 3.   Hooli inc [storeCert]
```

Comment:
We shouldn't use names not owned by Microsoft, even in docs/examples/specs. Use Fabrikam instead here, and line 134, below.

Reply:
done